raiden/transfer/mediated_transfer/initiator.py

```diff
@@ -240,13 +240,9 @@ def try_new_route(
     )
 
     for reachable_route_state in reachable_route_states:
-        forward_channel_id = reachable_route_state.forward_channel_id
-
-        candidate_channel_state = forward_channel_id and channelidentifiers_to_channels.get(
-            forward_channel_id
-        )
-
-        assert isinstance(candidate_channel_state, NettingChannelState)
+        candidate_channel_state = channelidentifiers_to_channels[
```

Comment:
Should we add an `assert` before that the key is in the dict?

Reply:
Before the change, we did an `isinstance` afterwards, which would have failed if there was no element. So we're not being any more strict than before.